app/channels/course/monitoring/live_monitoring_channel.rb

```diff
@@ -64,6 +64,9 @@ def active_sessions_snapshots
       is_valid_secret = @monitor.valid_secret?(last_heartbeat&.user_agent)
 
       course_user = course_users_hash[session.creator_id]
+      # This technically shouldn't happen, but can happen if someone is removed from
+      # the course after they finish a monitored assessment.
+      return [] unless course_user
```

Comment:
If `course_user` is `nil`, this causes `active_sessions_snapshots` to return an empty array instead of hashmap,  
Can use `next [nil, nil] unless course_user` instead, then do `hashmap.compact` at the end of the `active_sessions_snapshots` function to remove the nil entries.